### PR TITLE
First page : rootDirectory creation

### PR DESCRIPTION
**Pull request by souissimai**

### CODE REVIEW

File: src/components/directory-content.js [hunk 1]
```diff
@@ -9,8 +9,10 @@ import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 import { setActiveDirectory, setSelectionForCopy } from '../redux/actions';
 import { FormattedMessage, useIntl } from 'react-intl';
+import { DialogsId } from '../utils/UIconstants';
```
Comment (AAJELLAL) [inline src/components/directory-content.js]: You don't need this import. **constants** contains all elements . You can use it by calling  **constant.DialogsId**

File: src/components/directory-content.js [hunk 2]
```diff
@@ -821,7 +855,35 @@ const DirectoryContent = () => {
             })),
         [childrenMetadata, currentChildren, getElementTypeTranslation]
     );
+    const handleOpenDialog = useCallback(() => {
+        setOpenDialog(DialogsId.ADD_ROOT_DIRECTORY);
+    }, [setOpenDialog]);
 
+    const renderNoContent = () => {
+        return (
+            <>
```
Comment (AAJELLAL) [inline src/components/directory-content.js]: No need for this.

File: src/components/directory-content.js [hunk 3]
```diff
@@ -949,8 +1011,10 @@ const DirectoryContent = () => {
         }
 
         // If no selection or currentChildren = null (first time) render nothing
-        // TODO : Make a beautiful page here
         if (!currentChildren || !selectedDirectory) {
+            if (treeData.rootDirectories.length === 0 && treeData.initialized) {
```
Comment (AAJELLAL) [inline src/components/directory-content.js]: Why do you need the **initialized**? 
**treeData.rootDirectories.length === 0** is not sufficient in order to show renderNoContent?

File: src/components/tree-views-container.js [hunk 4]
```diff
@@ -287,6 +287,7 @@ const TreeViewsContainer = () => {
                     setTreeData({
                         rootDirectories: nrs,
                         mapData: mdr,
+                        initialized: true,
```
Comment (AAJELLAL) [inline src/components/tree-views-container.js]: the same here.

File: src/translations/fr.json [hunk 5]
```diff
@@ -17,6 +17,7 @@
   "private": "privé",
   "public": "public",
   "emptyDir": "Dossier vide",
+  "firstDir": "Créer  votre premier dossier",
```
Comment (AAJELLAL) [inline src/translations/fr.json]: Remove the extra space plz.

**Review by AAJELLAL — CHANGES_REQUESTED** [review]

File: src/components/directory-content.js (hunk 3 above)
Comment (souissimai) [reply src/components/directory-content.js]: yes need it because it takes time to fetch rootDirectories (as discussed)

File: src/components/directory-content.js [hunk 6]
```diff
@@ -821,7 +854,31 @@ const DirectoryContent = () => {
...
+                        variant="contained"
+                        sx={styles.noContentButton}
+                        onClick={handleOpenDialog}
+                    >
+                        <FormattedMessage id={'createFolder'} />
+                    </Button>
+                </Box>
+            </Box>
+        );
+    };
```
Comment (klesaulnier) [inline src/components/directory-content.js]: make a new component instead, in another file
this file is already big enough :)

File: src/components/directory-content.js [hunk 7]
```diff
@@ -821,7 +854,31 @@ const DirectoryContent = () => {
...
+    const renderNoContent = () => {
+        return (
+            <Box sx={styles.noContentContainer}>
+                <Box sx={styles.noContentCircle}>
+                    <CreateNewFolderOutlinedIcon sx={styles.noContentIcon} />
+                </Box>
+                <Box sx={styles.noContentText}>
+                    <h1>
+                        <FormattedMessage id={'firstDir'} />
+                    </h1>
```
Comment (klesaulnier) [inline src/components/directory-content.js]: this message is hard to see, change its color ?

File: src/components/directory-content.js [hunk 8]
```diff
@@ -98,6 +98,38 @@ const styles = {
...
+        flexDirection: 'column',
+        justifyContent: 'center',
+        alignItems: 'center',
+        marginTop: theme.spacing(10),
+    }),
+    noContentCircle: (theme) => ({
+        position: 'relative',
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
```
Comment (klesaulnier) [inline src/components/directory-content.js]: position: relative here and position: absolute for child component are doing the job, we don't need those 3 attributes

File: src/components/directory-content.js [hunk 9]
```diff
@@ -98,6 +98,38 @@ const styles = {
         maxWidth: '250px',
         maxHeight: '50px',
     },
+    noContentContainer: (theme) => ({
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'center',
+        alignItems: 'center',
```
Comment (klesaulnier) [inline src/components/directory-content.js]: I don't think we need "alignItems: 'center'"

File: src/components/directory-content.js [hunk 10]
```diff
@@ -98,6 +98,38 @@ const styles = {
...
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+        width: 200,
+        height: 200,
+        backgroundColor: theme.row.primary,
+        borderRadius: theme.spacing(15),
+    }),
+    noContentIcon: {
+        fontSize: '100px',
```
Comment (klesaulnier) [inline src/components/directory-content.js]: I think I would make a component which only contain this icon is a circle

Then I would create a constant "CIRCLE_SIZE"
Then here, width and height would be equal to "CIRCLE_SIZE"
"fontSize" would be equals to "CIRCLE_SIZE / 2"

I would make it easier to reuse and isolate complex CSS

File: src/components/directory-content.js [hunk 11]
```diff
@@ -766,7 +799,7 @@ const DirectoryContent = () => {
                 });
         }
         setSelectedUuids(new Set());
-    }, [handleError, currentChildren, currentChildrenRef]);
+    }, [handleError, currentChildren, currentChildrenRef, treeData]);
```
Comment (klesaulnier) [inline src/components/directory-content.js]: which is this change necessary ?

File: src/components/directory-content.js [hunk 12]
```diff
@@ -821,7 +854,31 @@ const DirectoryContent = () => {
             })),
         [childrenMetadata, currentChildren, getElementTypeTranslation]
     );
+    const handleOpenDialog = useCallback(() => {
+        setOpenDialog(constants.DialogsId.ADD_ROOT_DIRECTORY);
+    }, [setOpenDialog]);
```
Comment (klesaulnier) [inline src/components/directory-content.js]: since "setOpenDialog" is a "useState" setter, you don't need to put it in dependencies

File: src/translations/en.json [hunk 13]
```diff
@@ -17,6 +17,7 @@
   "private": "private",
   "public": "public",
   "emptyDir": "Empty directory",
+  "firstDir": "Create your first directory",
```
Comment (klesaulnier) [inline src/translations/en.json]:
```suggestion
  "createFirstDir": "Create your first directory",
```

File: src/components/directory-content.js (hunk 3 above)
Comment (klesaulnier) [reply src/components/directory-content.js]: I think is would be cleaner with a "isLoading" functionnality, but it might be a little complex since treeData is is redux
here it sounds like a hack

**Review by klesaulnier — CHANGES_REQUESTED** [review]

File: src/components/directory-content.js (hunk 3 above)
Comment (souissimai) [reply src/components/directory-content.js]: at the beginning I was thinking about adding isLoading functionality, but didnt find another way to do it since treedata is redux

File: src/components/directory-content.js (hunk 12 above)
Comment (souissimai) [reply src/components/directory-content.js]: done

File: src/components/directory-content.js (hunk 7 above)
Comment (souissimai) [reply src/components/directory-content.js]: color: theme.row.primary what do you suggest, comparing it with the provided image of stephane it's the same color desired.

File: src/components/directory-content.js (hunk 11 above)
Comment (souissimai) [reply src/components/directory-content.js]: unuseful to clean

File: src/components/directory-content.js (hunk 9 above)
Comment (souissimai) [reply src/components/directory-content.js]: Ithink we need it to make it on the center : image

File: src/translations/en.json (hunk 13 above)
Comment (souissimai) [reply src/translations/en.json]: done

File: src/components/directory-content.js (hunk 6 above)
Comment (souissimai) [reply src/components/directory-content.js]: done, refactored

File: src/components/directory-content.js (hunk 10 above)
Comment (souissimai) [reply src/components/directory-content.js]: Done

**Review by AAJELLAL — APPROVED** [review]
Tests and code are OK.
No warnings in console.

File: src/components/no-content-directory.tsx [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+        noContentButton: {
+            borderRadius: '30px',
+        },
+    };
+
+    return (
+        <Box sx={styles.noContentContainer}>
+            <CircleIcon size={CIRCLE_SIZE} iconStyles={stylesIcon.circle}>
+                <CreateNewFolderOutlinedIcon
+                    style={{ fontSize: CIRCLE_SIZE / 2 }}
```
Comment (klesaulnier) [inline src/components/no-content-directory.tsx]: use sx instead since it's MUI Icon

File: src/components/no-content-directory.tsx [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+            alignItems: 'center',
+            marginTop: theme.spacing(20),
+        }),
+        noContentIcon: {
+            fontSize: `${CIRCLE_SIZE / 2}px`,
+            position: 'absolute',
+            top: '50%',
+            left: '50%',
+            transform: 'translate(-50%, -50%)',
+        },
```
Comment (klesaulnier) [inline src/components/no-content-directory.tsx]: unused class, to remove

**Review by klesaulnier — APPROVED** [review]
still confused about the colors not being very easy to read, but nothing that can't be change easily

Test: OK
Code: OK

File: src/components/no-content-directory.tsx (hunk 15 above)
Comment (souissimai) [reply src/components/no-content-directory.tsx]: done

File: src/components/no-content-directory.tsx (hunk 14 above)
Comment (souissimai) [reply src/components/no-content-directory.tsx]: done
